File: foody-master/Front End/src/app/services/cart.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { CartItem } from '../model/cart-item';
import { cartUrl } from '../config/api';
import { map } from 'rxjs/operators';
import { FoodService } from './food.service';

@Injectable({
  providedIn: 'root'
})
export class CartService {

  cartItem: CartItem;
  cartId: number = 1;
  results: number;
  constructor(
    private http: HttpClient
  ) { }

  getCartItems():Observable<CartItem[]>{
    return this.http.get<CartItem[]>(cartUrl);
  }

  getCartById(id: number):Observable<CartItem>{
    return this.http.get<CartItem>(cartUrl+"/"+id);
  }
  newCart(cart: CartItem){
    return this.http.post(cartUrl,cart);
  }
}

File: foody-master/Back End/src/main/java/com/food/service/CartService.java
package com.food.service;

import java.util.List;

import com.food.model.Cart;

public interface CartService {
	public Cart addCart(Cart cart);
	public Cart updateCart(Cart cart);
	public Cart getCartById(int id);
	public List<Cart> getAllCarts();
	public void removeCart(int id);
}
